spec/models/miq_expression_spec.rb

```diff
@@ -1631,4 +1631,140 @@
       end
     end
   end
+
+  describe "#sql_supports_atom?" do
+    context "expression key is 'CONTAINS'" do
+      context "operations with 'tag'" do
+        it "returns true for tag of the main model" do
+          expression = {"CONTAINS" => {"tag" => "VmInfra.managed-operations", "value" => "analysis_failed"}}
+          expect(described_class.new(nil).sql_supports_atom?(expression)).to eq(true)
+        end
+
+        it "returns false if for tag of not main model" do
```

Comment:
This should be reworded for clarity. Something like "returns false for tag of associated model"